src/clients/Deluge.ts

```diff
@@ -140,34 +168,38 @@ export default class Deluge implements TorrentClient {
 	 * parses the set-cookie header and updates stored value
 	 */
 	private handleResponseHeaders(headers: Headers) {
-		if (headers.has("Set-Cookie")) {
-			this.delugeCookie = headers.get("Set-Cookie").split(";")[0];
-		}
+		this.delugeCookie = headers.has("Set-Cookie")
+			? headers.get("Set-Cookie").split(";")[0]
+			: null;
 	}
 
 	/**
 	 * checks enabled plugins for "Label"
 	 * returns true if successful.
 	 */
 	private async labelEnabled() {
-		const enabledLabels = await this.call("core.get_enabled_plugins", []);
-		return enabledLabels?.result?.includes?.("Label");
+		const enabledPlugins = await this.call<string>(
+			"core.get_enabled_plugins",
+			[]
+		);
+		return enabledPlugins ? enabledPlugins.result.includes("Label") : false;
```

Comment:
I think what you're really looking for here is this:
```suggestion
		return enabledPlugins.result ? enabledPlugins.result.includes("Label") : false;
```

AFAIK, the return value of `call` will always be an object, it's the result and error properties _on_ that object that may or not be there

Reply:
It will either throw or return JSON response as `Promise<DelugeJSON<ResultType>>` - there are no other possibilities as far as I remember. I can double-check.